src/main/java/com/squareup/javapoet/AnnotationSpec.java

```diff
@@ -98,6 +104,63 @@ private void emitAnnotationValues(CodeWriter codeWriter, String whitespace,
     codeWriter.emit(whitespace + "}");
   }
 
+  public static AnnotationSpec get(Annotation annotation) {
+    return get(annotation, false);
+  }
+
+  public static AnnotationSpec get(Annotation annotation, boolean includeDefaultValues) {
+    Builder builder = builder(annotation.annotationType());
+    try {
+      Method[] methods = annotation.annotationType().getDeclaredMethods();
+      Arrays.sort(methods, new MethodNameComparator());
+      for (Method method : methods) {
+        Object value = method.invoke(annotation);
+        if (!includeDefaultValues) {
```

Comment:
...ahh, I see. if the annotation reports a default value you can omit it. Hmm. I think it's simpler to just universally include everything?